core/trino-main/src/test/java/io/trino/operator/aggregation/TestMultimapAggAggregation.java

```diff
@@ -88,6 +88,13 @@ public void testNullMap()
         testMultimapAgg(DOUBLE, ImmutableList.<Double>of(), VARCHAR, ImmutableList.<String>of());
     }
 
+    @Test
+    public void testKeysUseIsDistinctSemantics()
+    {
+        testMultimapAgg(DOUBLE, ImmutableList.of(Double.NaN, Double.NaN), BIGINT, ImmutableList.of(1L, 1L));
```

Comment:
this testing sucks; we should explicity compare with expected multimap. Not construct it risking same bug in constructing and tested code. (no changes requested)

Reply:
Agreed.